specs/differences-from-optimism-bedrock.md

```diff
@@ -23,33 +23,57 @@
+checkpoint outputs from `sequencers` and assigned it to `verifiers`. As a result, these participants have been renamed
+to reflect these role changes and the future direction of Kroma's decentralization. For more detailed information about
+our decentralization scheme, please refer to
+[this article](https://medium.com/@kroma-network/the-road-to-kromas-decentralization-38f8e46df442)
+on the Kroma blog.
 
 ### Sequencer -> Proposer
 
-We use `sequencing` for transaction inclusion, exclusion and ordering. Currently, `sequencer` does not only
-this but also does block building. Thinking of Ethereum PoS, this is what `proposer` exactly does, which is building
-and proposing a block. To take this into consideration, we rename the actor `sequencer` to `proposer` and plan to
-separate block building ability from `proposer` in the future. We will make the results of our research public.
+We use *sequencing* for the inclusion, exclusion and ordering of transactions. Currently, the `sequencer`
+not only performs these functions but also proposes built blocks to Layer 1. In Ethereum PoS, the `proposer` is
+responsible for precisely what the `sequencer` does - building and proposing a block. With this in mind, we have decided
+to rename the actor from `sequencer` to `proposer` and intend to separate the block building capability from the
+proposer in the future. We are comitted to sharing the results of our research with the public.
+
+### Verifier -> Validator
+
+We utilize the term `validator` to denote a participant who is responsible for submitting the
+[L2 output root][g-l2-output-root] and validating its accuracy by either submitting dispute challenges (during the
+optimistic rollup phase) or providing zk validity proofs (during the zk rollup phase). This concept bears a resemblance
+to how L1 validators cast FFG votes at each epoch.
 
-### Proposer -> Validator
+### Compositions
 
-For the same reason above, we use the term `validator` to represent an entity that submits
-[L2 output root][g-l2-output-root]. Because this resembles how L1 validator casts FFG vote at every epoch.
+Kroma maintains the modular architecture of the OP Stack, with various components communicating through Json RPC calls.
+As part of the transition from `verifier` to `validator`, we have renamed the `proposer` (op-proposer) component of the
+OP Stack to `validator` (kroma-validator) and made necessary modifications to the code to handle the dispute challenge
+processes.
+
+The followings are components that are used to run different types of nodes:
+
+| Node        | Components                                                           |
+|-------------|----------------------------------------------------------------------|
+| `Proposer`  | `L2 EL client` + `L2 CL client` + `kroma-batcher`                    |
+| `Validator` | `L2 EL client` + `L2 CL client` + `kroma-validator` + `kroma-prover` |
```

Comment:
`kroma-prover` might be optional, isn't it?
```suggestion
| `Validator` | `L2 EL client` + `L2 CL client` + `kroma-validator` (+ optionally `kroma-prover`) |
```

Reply:
I itended to remove 'optional'. We need to talk about this but let's move optional at this time.